test/extended/etcd/helpers/helpers.go

```diff
@@ -180,6 +182,78 @@ func recoverClusterToInitialStateIfNeeded(ctx context.Context, t TestingT, machi
 	})
 }
 
+func DeleteSingleMachine(ctx context.Context, t TestingT, machineClient machinev1beta1client.MachineInterface) (string, error) {
+	waitPollInterval := 15 * time.Second
+	waitPollTimeout := 5 * time.Minute
+	t.Logf("Waiting up to %s to delete a machine", waitPollTimeout.String())
+
+	machineToDelete := ""
+	err := wait.Poll(waitPollInterval, waitPollTimeout, func() (bool, error) {
+		machineList, err := machineClient.List(ctx, metav1.ListOptions{LabelSelector: masterMachineLabelSelector})
+		if err != nil {
+			return isTransientAPIError(t, err)
+		}
+
+		// No need to be deterministic since it doesn't matter which machine we delete
+		machineToDelete = machineList.Items[0].Name
```

Comment:
there might be two invocations of the wait.Poll, especially if the delete failed on some transient client/reponse error. So we _could_ end up with two deleted masters here. 

I would just remove the wait.Poll altogether and rely on the machineClient retries on a single chosen machine name.

Reply:
Good catch. I'm going to make it deterministic by always picking the master name with the lowest index since all machines are suffixed e.g `...-master-0` `...-master-1` etc.

> remove the wait.Poll altogether and rely on the machineClient retries

The wait.Poll is for retrying, so not sure if you meant replacing it with another util? The machineClient won't auto retry. Or do you mean just loop the Delete until it goes through. Would need some backoff though in case of transient API errors which is what the wait.Poll does.

Reply:
ahh, thank you. I thought that there's some retry utility baked into the clients already.  As long as you pick the master deterministically it's fine :)